plugins/woocommerce-admin/client/task-lists/fills/tax/index.tsx

```diff
@@ -174,39 +177,43 @@ export const Tax: React.FC< TaxProps > = ( { onComplete, query, task } ) => {
 					! woocommerceShippingActivated &&
 					woocommerceTaxCountries.includes( countryCode ),
 			},
-		];
+			{
+				id: 'stripe-tax',
+				card: StripeTaxCard,
 
-		return partners.filter( ( partner ) => partner.isVisible );
-	};
+				isVisible: stripeTaxCountries.includes( countryCode ),
+			},
+		];
 
-	const partners = getVisiblePartners();
+		return allPartners.filter( ( partner ) => partner.isVisible );
+		// eslint-disable-next-line react-hooks/exhaustive-deps -- the partner list shouldn't be changing in the middle of interaction. for some reason the country is becoming null in a re-render and causing unexpected behaviour
+	}, [] );
 
+	const { auto } = query;
 	useEffect( () => {
-		const { auto } = query;
-
 		if ( auto === 'true' ) {
 			onAutomate();
-			return;
 		}
+	}, [ auto, onAutomate ] );
 
+	useEffect( () => {
 		if ( query.partner ) {
 			return;
 		}
-
 		recordEvent( 'tasklist_tax_view_options', {
 			options: partners.map( ( partner ) => partner.id ),
 		} );
-	}, [ onAutomate, partners, query ] );
+	}, [ partners, query ] );
 
-	const getCurrentPartner = () => {
+	const currentPartner = useMemo( () => {
 		if ( ! query.partner ) {
 			return null;
 		}
 
 		return (
 			partners.find( ( partner ) => partner.id === query.partner ) || null
 		);
-	};
+	}, [ partners, query ] );
```

Comment:
```suggestion
	}, [ partners, query.partner ] );
```

Same suggestion as above

Reply:
I think this makes sense!